test/import/loadConfigurationFromIgResource.test.ts

```diff
@@ -0,0 +1,105 @@
+    expect(config).toEqual({ canonical: 'http://example.org', FSHOnly: true, fhirVersion: [] });
+    expect(loggerSpy.getLastMessage('info')).toMatch(
+      new RegExp(
+        `from ${escapeRegExp(
+          path.join(inputPath, '..', 'sneaky-input', 'ImplementationGuide.json')
+        )}`
+      )
+    );
+  });
+
+  it('should extract a configuration with only a url', () => {
+    const inputPath = path.join(__dirname, 'fixtures', 'ig-JSON-only-url', 'fsh');
+    const config = loadConfigurationFromIgResource(inputPath);
+    expect(config).toEqual({ canonical: 'http://example.org', FSHOnly: true, fhirVersion: [] });
+    expect(loggerSpy.getLastMessage('info')).toMatch(
+      new RegExp(
+        `from ${escapeRegExp(path.join(inputPath, '..', 'input', 'ImplementationGuide.json'))}`
+      )
+    );
+  });
+
+  it('should extract a configuration with a url and dependencies', () => {
+    const inputPath = path.join(__dirname, 'fixtures', 'ig-JSON', 'fsh');
+    const config = loadConfigurationFromIgResource(inputPath);
+    expect(config).toEqual({
+      canonical: 'http://example.org',
+      FSHOnly: true,
+      dependencies: [
+        { packageId: 'foo.bar', version: '1.2.3' },
+        { packageId: 'bar.foo', version: 'current' }
+      ],
+      fhirVersion: ['4.0.1']
+    });
+    expect(loggerSpy.getLastMessage('info')).toMatch(
+      new RegExp(
+        `from ${escapeRegExp(path.join(inputPath, '..', 'input', 'ImplementationGuide.json'))}`
+      )
+    );
+  });
+
+  it('should extract an XML configuration with a url and dependencies', () => {
+    const inputPath = path.join(__dirname, 'fixtures', 'ig-XML', 'fsh');
+    const config = loadConfigurationFromIgResource(inputPath);
+    expect(config).toEqual({
+      canonical: 'http://example.org',
+      FSHOnly: true,
+      dependencies: [
+        { packageId: 'foo.bar', version: '1.2.3' },
+        { packageId: 'bar.foo', version: 'current' }
+      ],
+      fhirVersion: ['4.0.1']
+    });
+    expect(loggerSpy.getLastMessage('info')).toMatch(
+      new RegExp(
+        `from ${escapeRegExp(path.join(inputPath, '..', 'input', 'ImplementationGuide.xml'))}`
+      )
+    );
+  });
+
+  it('should find the ImplementationGuide JSON file even when other files are present', () => {
+    const inputPath = path.join(__dirname, 'fixtures', 'ig-JSON-other-files', 'fsh');
+    const config = loadConfigurationFromIgResource(inputPath);
+    expect(config).toEqual({ canonical: 'http://example.org', FSHOnly: true, fhirVersion: [] });
+    expect(loggerSpy.getLastMessage('info')).toMatch(
+      new RegExp(
+        `from ${escapeRegExp(path.join(inputPath, '..', 'input', 'ImplementationGuide.json'))}`
+      )
+    );
+  });
+
+  it('should return null when there are two potentially valid IG files', () => {
+    const inputPath = path.join(__dirname, 'fixtures', 'ig-JSON-and-ig-XML', 'fsh');
+    expect(loadConfigurationFromIgResource(inputPath)).toBeNull();
```

Comment:
I'm not actually sure that `null` will get returned if there is both a JSON and XML representation of the IG. I think the input path here should actually be `ig-XML-and-ig-JSON`. When I do switch that, the test fails. I think it might be because the `return null` in [loadConfigurationFromIgResource line 46](https://github.com/FHIR/sushi/pull/577/files#diff-41eb50e93ad9d6ea6e77945583b53084R46) just moves on to the next iteration of the forEach`.

Reply:
Good catch, in 359d325 I made the test actually test things, and fixed the code so that the case the test is testing for passes.

Reply:
Looks good to me! The only other minor thought I had if you think it makes sense to log a message that multiple files were found and that's why the information can't be extracted. Right now I just get

```

Reply:
I don't feel strongly about this if we don't want to have two messages and it's complicated to have the one message be conditional.

Reply:
Ugh.  And it did it to me again.  I thought I submitted the above comment _yesterday_ -- but it just added it to my pending comments instead (even though it _looked_ like it submitted it).  Sorry!

Reply:
I hadn't seen Julia's further comments. I'll improve the error message.

Reply:
Error message added in https://github.com/FHIR/sushi/pull/577/commits/32d3efdd47a26e22ce6ae69a286c8c66f44c0fd0